### ISSUE TITLE
Putting all nested views into one ui-view

### ISSUE THREAD

**Issue by archasek** [angular-ui/ui-router#3000]
Is it possible?
I just want to put all my views without names into one ui-view. Unfortunately, only the first view is injected, not all.

```
        $stateProvider
            .state('root.home', {
                url: '/',
                views: {
                    'main@': {
                        template: '<ui-view/>'
                    },
                    '@root.home': {
                        component: 'componentXXX'
                    },
                    '@root.home': {
                        component: 'componentYYY'
                    }
                }
            });
```

**Comment by christopherthielen** [comment]
I don't understand what you mean by "put all my views without names into one ui-view".  Can you rephrase what you're trying to accomplish, and show an example of what the markup would look like?

---

You can indeed create a `main@` with a layout (like your '<ui-view/>' example), then target the layout from another view like your example ('@root.home': {}). 

But there are two problems with your example:

1) `<ui-view/>` should be `<ui-view></ui-view>`
2) You can't have two keys in a javascript object that share the same name (you have two '@root.home' keys)

**Comment by archasek** [comment]
Thanks for the answer. The 2. point is the one I was asking for (a few views that are unnamed).

**Comment by christopherthielen** [comment]
Well as far as I understand your question, no, its not possible.  This is what named views are for.

May I  suggest creating a parent component that renders the child components

```
.component('home', {
  bindings: { xdata: '<', ydata: '<' },
  template: `
    <componentXXX xdata="xdata"></componentXXX>
    <componentYYY ydata="ydata"></componentYYY>
  `
})
```

**Comment by christopherthielen** [comment]
I'm closing because I think your question has been answered?

**Comment by archasek** [comment]
Yes, thank you!
